frontend/model/contracts/group.js

```diff
@@ -1073,6 +1092,13 @@ sbp('chelonia/defineContract', {
           Vue.set(state.settings, key, data[key])
         }
 
+        if ('distributionDate' in data) {
+          Vue.set(state, 'paymentsByPeriod', {})
+          const curPeriodPayments = initFetchPeriodPayments({ contractID, meta, state, getters })
+          const period = Object.keys(getters.groupPeriodPayments)[0]
+          curPeriodPayments.haveNeedsSnapshot = getters.haveNeedsForThisPeriod(period)
+        }
```

Comment:
> It is only safe to update the distribution date if meta.createdDate is before the current distribution date. So you need to first create an if condition here to check for that.

I think I did it in the `validate` function.
```
else if ('distributionDate' in data &&
    (getters.groupDistributionStarted(meta.createdDate) || Object.keys(getters.groupPeriodPayments).length > 1)) {
    throw new TypeError(L('Can\'t change distribution date because distribution period has already started.'))
}
```